metals/src/main/scala/scala/meta/internal/bsp/BspExtra.scala

```diff
@@ -0,0 +1,120 @@
+package scala.meta.internal.bsp
+
+import scala.jdk.CollectionConverters._
+
+import scala.meta.internal.metals.BuildInfo
+import scala.meta.internal.semver.SemVer
+
+/**
+ * When Metals establish bsp-connection it sends
+ *  the required semanticdb version and supportedScalaVersions (only scala2) to bsp-server
+ *  to prepare compiler plugin setup on its side.
+ *
+ * To have the actual state of this metadata use `BspExtra.discoverNewReleases`.
+ * It checks if there were scalameta/semanticdb releases that support newer Scala2 versions
+ *  that were published after the Metals release.
+ */
+final case class BspExtra(
+    semanticdbVersion: String,
+    supportedScalaVersions: java.util.List[String]
+)
+
+object BspExtra {
+
+  def apply(
+      semanticdbVersion: String,
+      supportedScalaVersions: Seq[String]
+  ): BspExtra =
+    BspExtra(semanticdbVersion, supportedScalaVersions.asJava)
+
+  val knownAtReleaseMoment: BspExtra =
+    BspExtra(
+      BuildInfo.semanticdbVersion,
+      BuildInfo.supportedScala2Versions
+    )
+
+  val allowedSemanticdbVersionPrefix = "4.4"
+
+  def discoverNewReleases(): BspExtra =
+    discoverNewReleases(knownAtReleaseMoment)
+
+  def discoverNewReleases(known: BspExtra): BspExtra = {
+
+    def semanticdbScalacPluginVersion(
+        scalaVersion: String
+    ): Option[SemVer.Version] = {
+      val versionsQuery =
+        s"org.scalameta:semanticdb-scalac_$scalaVersion:"
+
+      val completions = coursierapi.Complete
+        .create()
+        .withInput(versionsQuery)
+        .complete()
+        .getCompletions()
+        .asScala
+
+      completions
+        .filter(_.startsWith(allowedSemanticdbVersionPrefix))
+        .map(SemVer.Version.fromString)
+        .filter(_.isStable)
+        .sortWith(_ > _)
+        .headOption
```

Comment:
Shouldn't it be wrapped in `Future`? It's blocking, it's doing an API call. Maybe it's better to explicitly wrap it into the future rather than expecting call site to do it?

Reply:
> It's blocking

I know :smiling_face_with_tear:
Just decided to leave it as it is because it's how we use coursier api in other places. 

Probably, it's a bit late to care blocking stuff in Metals. Future in Metals codebase appears only when there is a real need for it. And I think that's ok.  In case of LSP server,  it doesn't have to serve thousands requests per seconds so we can allow us writing things ignoring these bloking/async matters